mu/logic.py

```diff
@@ -112,12 +111,45 @@ def get_settings_path():
     if not os.path.exists(settings_dir):
         settings_dir = os.path.join(DATA_DIR, settings_filename)
         if not os.path.exists(settings_dir):
-            with open(settings_dir, 'w') as f:
-                logger.debug('Creating settings file: {}'.format(settings_dir))
-                json.dump({}, f)
+            try:
+                with open(settings_dir, 'w') as f:
+                    logger.debug('Creating settings file: {}'.format(
+                                 settings_dir))
+                    json.dump({}, f)
+            except FileNotFoundError:
+                logger.error('Unable to create settings file: {}'.format(
+                             settings_dir))
     return settings_dir
 
 
+def get_python_dir():
+    """
+    The default is to use a directory in the users home folder however
+    in some network systems this in inaccessable. This allows a key in the
+    settings file to be used to set a custom path
+    """
+    settings_path = get_settings_path()
+    try:
+        with open(settings_path) as f:
+            try:
+                # Load up the JSON
+                sets = json.load(f)
+                if 'python_dir' in sets:
+                    # They have set a custom folder return that
```

Comment:
Sorry about being pedantic about a comment, but technically since we are writing `python_dir` into the settings file then finding this key would be expected in most cases, so it doesn't really indicate a custom path has been set.